src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -356,6 +362,17 @@ private void HandleChangesForStaticIndexes(DatabaseRecord record, long index, Li
             }
         }
 
+        private bool ShouldSkipThisNode(DatabaseRecord record, string indexName)
+        {
+            if (record.StaggeredIndexes.TryGetValue(indexName, out var staggeredIndexesByNodeTag) == false)
+                return false; // Can this happen? Should we throw or ignore?
+
+            if (staggeredIndexesByNodeTag.TryGetValue(_serverStore.NodeTag, out var staggeredIndexStatus) == false)
+                return false; // Can this happen? Should we throw or ignore?
```

Comment:
Yes, can happen, when we add a node to the database while the stagerred index is running.
Note that we need to decide what to do here, I'm leaning toward starting this immediately, instead.

Reply:
Yes, but the comment should be clearer